curriculum/challenges/english/25-front-end-development/lab-markdown-to-html-converter/66f55eac933ff64ce654ca74.md

```diff
@@ -0,0 +1,850 @@
+assert.lengthOf(quotes, 1);
+assert.lengthOf(quotes[0].children, 1);
+assert.lengthOf(strongs, 1);
+assert.lengthOf(strongs[0].children, 1);
+assert.equal(strongs[0].innerText, "this is a quote");
+assert.lengthOf(italics, 1);
+assert.equal(italics[0].innerText, "quote");
+```
+
+You should have only one `script` element in your HTML.
+
+```js
+assert.lengthOf(document.querySelectorAll("script"), 3);
+```
+
+# --seed--
+
+## --after-user-code--
+
+```js
+const input = document.querySelector("#markdown-input");
+const output = document.querySelector("#html-output");
+const preview = document.querySelector("#preview");
+```
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Markdown to HTML Converter</title>
+    <link rel="stylesheet" href="styles.css">
+</head>
+
+<body>
+    <h1>Markdown to HTML Converter</h1>
+    <div id="container">
+        <div class="container">
+            <h2>Markdown Input:</h2>
+            <textarea id="markdown-input" placeholder="Enter your markdown here..."></textarea>
+        </div>
+        <div class="container">
+            <h2>Raw HTML Output:</h2>
+            <div id="html-output"></div>
+        </div>
+        <div class="container">
+            <h2>HTML Preview:</h2>
+            <div id="preview"></div>
+        </div>
+    </div>
+</body>
+
+</html>
+```
+
+```css
+* {
+     box-sizing: border-box;
+}
+ body {
+     font-family: Arial, sans-serif;
+     padding: 20px;
+}
+ #markdown-input {
+     width: 100%;
+     height: 100px;
+}
+ #html-output, #preview {
+     height: 100px;
+     display: inline-block;
+     width: 100%;
+     border: 1px solid #ccc;
+     padding: 10px;
+     margin: auto;
+     white-space: pre-wrap;
+     background-color: #f9f9f9;
+}
+ @media (min-width: 600px) {
+     #markdown-input, #html-output, #preview {
+         height: 200px;
+         margin: 0;
+    }
+     #container {
+         display: flex;
+         justify-content: space-evenly;
+         gap: 10px;
+    }
+}
+```
+
+```js
+
+```
+
+# --solutions--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Markdown to HTML Converter</title>
+    <link rel="stylesheet" href="styles.css">
+</head>
+
+<body>
+    <h1>Markdown to HTML Converter</h1>
+    <div id="container">
+        <div class="container">
+            <h2>Markdown Input:</h2>
+            <textarea id="markdown-input" placeholder="Enter your markdown here..."></textarea>
+        </div>
+        <div class="container">
+            <h2>Raw HTML Output:</h2>
+            <div id="html-output"></div>
+        </div>
+        <div class="container">
+            <h2>HTML Preview:</h2>
+            <div id="preview"></div>
+        </div>
+    </div>
+    <script src="script.js"></script>
```

Comment:
It's not mentioned that this is not included in the seed, it seems you can start working on the JS directly

Reply:
There's a note at the bottom:
`**Note:** Be sure to link your JavaScript file in your HTML.`